src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1562,11 +1567,22 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			this._initDimensions();
 			this.xterm?.raw.resize(this._cols || Constants.DefaultCols, this._rows || Constants.DefaultRows);
 		}
-
 		const originalIcon = this.shellLaunchConfig.icon;
-		await this._processManager.createProcess(this._shellLaunchConfig, this._cols || Constants.DefaultCols, this._rows || Constants.DefaultRows, this._accessibilityService.isScreenReaderOptimized()).then(error => {
+		await this._processManager.createProcess(this._shellLaunchConfig, this._cols || Constants.DefaultCols, this._rows || Constants.DefaultRows, this._accessibilityService.isScreenReaderOptimized()).then(async error => {
 			if (error) {
-				this._onProcessExit(error, error.code === ShellIntegrationExitCode);
+				if (this._usedShellIntegrationInjection) {
+					await this.reuseTerminal({ ignoreShellIntegration: true });
+					this.statusList.add({
+						id: TerminalStatus.ShellIntegrationAttentionNeeded,
+						severity: Severity.Warning,
+						icon: Codicon.warning,
+						tooltip: nls.localize('launchFailed.exitCodeOnlyShellIntegration', "The terminal process failed to launch (exit code: {0}). Disabling shell integration with `terminal.integrated.shellIntegration.enabled` might help.", error.code),
```

Comment:
I know we aren't keen on people outright disabling this, but if it's causing the process to exit, we need to tell them why IMO.

Reply:
so that on the next terminal launch, they know what to do.

could also add a link to docs?

Reply:
Telling how to disable when in a broken state is fine, we just don't want to do it automatically.